spec-main/api-autoupdater-darwin-spec.ts

```diff
@@ -0,0 +1,263 @@
+
+describeFn('autoUpdater behavior', function () {
+  this.timeout(120000)
+
+  let identity = ''
+
+  beforeEach(function () {
+    const result = cp.spawnSync('bash', [path.resolve(__dirname, '../script/codesign/get-trusted-identity.sh')])
+    if (result.status !== 0 || result.stdout.toString().trim().length === 0)  {
+      if (isCI) {
+        throw new Error('No valid signing identity available to run autoUpdater specs')
+      }
+      this.skip()
+    } else {
+      identity = result.stdout.toString().trim()
+    }
+  })
+
+  it('should have a valid code signing identity', () => {
+    expect(identity).to.be.a('string').with.lengthOf.at.least(1)
+  })
+
+  const copyApp = async (newDir: string, fixture = 'initial') => {
+    const appBundlePath = path.resolve(process.execPath, '../../..')
+    const newPath = path.resolve(newDir, 'Electron.app')
+    cp.execSync(`cp -R "${appBundlePath}" "${path.dirname(newPath)}"`)
+    const appDir = path.resolve(newPath, 'Contents/Resources/app')
+    await fs.mkdirp(appDir)
+    await fs.copy(path.resolve(fixturesPath, 'auto-update', fixture), appDir)
+    const plistPath = path.resolve(newPath, 'Contents', 'Info.plist')
+    await fs.writeFile(
+      plistPath,
+      (await fs.readFile(plistPath, 'utf8')).replace('<key>BuildMachineOSBuild</key>', `<key>NSAppTransportSecurity</key>
+      <dict>
+          <key>NSAllowsArbitraryLoads</key>
+          <true/>
+          <key>NSExceptionDomains</key>
+          <dict>
+              <key>localhost</key>
+              <dict>
+                  <key>NSExceptionAllowsInsecureHTTPLoads</key>
+                  <true/>
+                  <key>NSIncludesSubdomains</key>
+                  <true/>
+              </dict>
+          </dict>
+      </dict><key>BuildMachineOSBuild</key>`)
+    )
+    return newPath
+  }
+
+  const spawn = async (cmd: string, args: string[], opts: any = {}) => {
+    let out = ''
+    const child = cp.spawn(cmd, args, opts)
+    child.stdout.on('data', (chunk: Buffer) => {
+      out += chunk.toString()
+    })
+    child.stderr.on('data', (chunk: Buffer) => {
+      out += chunk.toString()
+    })
+    return await new Promise<{ code: number, out: string }>((resolve) => {
+      child.on('exit', (code, signal) => {
+        expect(signal).to.equal(null)
+        resolve({
+          code: code!,
+          out
+        })
+      })
+    })
+  }
+
+  const signApp = (appPath: string) => {
+    return spawn('codesign', ['-s', identity, '--deep', '--force', appPath])
+  }
+
+  const launchApp = async (appPath: string, args: string[] = []) => {
+    return spawn(path.resolve(appPath, 'Contents/MacOS/Electron'), args)
+  }
+
+  const withTempDirectory = async (fn: (dir: string) => Promise<void>) => {
+    const dir = await fs.mkdtemp(path.resolve(os.tmpdir(), 'electron-update-spec-'))
+    try {
+      await fn(dir)
+    } catch (err) {
+      cp.execSync(`rm -r "${dir}"`)
```

Comment:
```suggestion
      cp.execSync("rm", ["-rf", dir])
```

Reply:
Same as above, that's not how `exec` works 👍